frontend/ui/elements/common_settings_menu_table.lua

```diff
@@ -579,6 +604,39 @@ common_settings.document_metadata_location = {
         },
         genMetadataFolderMenuItem("doc"),
         genMetadataFolderMenuItem("dir"),
+        genMetadataFolderMenuItem("hash"),
+        { -- hash-based metadata count / TextViewer
+            text_func = function()
+                local hash_text = _("Show documents with hash-based metadata")
+                local no_hash_text = _("No documents with hash-based metadata")
+                if DocSettings.isHashLocationEnabled() then
+                    if G_reader_settings:readSetting("document_metadata_folder") ~= "hash" then
+                        return  "⚠ "..hash_text
+                    end
+                    return  hash_text
+                end
+                return no_hash_text
+            end,
+            keep_menu_open = true,
+            enabled_func = function()
+                return DocSettings.isHashLocationEnabled()
+            end,
+            callback = function()
+                local hash_file_infos = DocSettings.getHashDirSdrInfos()
+                local book_info_items = {}
+                for i, file_info in ipairs(hash_file_infos) do
+                    table.insert(book_info_items, table.concat({"\n", i, ". ", file_info}))
+                end
+                local book_info_str = table.concat(book_info_items)
+                UIManager:show(require("ui/widget/textviewer"):new{
+                    title = T(N_("1 document with hash-based metadata", "%1 documents with hash-based metadata",
+                        #hash_file_infos), #hash_file_infos),
```

Comment:
Nit but I find that illegible. ASCII art properly or not at all please. :-P
```suggestion
                    title = T(N_("1 document with hash-based metadata", "%1 documents with hash-based metadata", #hash_file_infos), #hash_file_infos),
```

Reply:
If long lines of code are okay, fine by me :)

Reply:
As I wrote in the style guide. ;-)

> Line Length - 100 characters as a rule of thumb, but please don't make code uglier and harder to grep with forced linebreaks to stick to an artificial limit